### PR TITLE
Query interface

### PR DESCRIPTION
**Pull request by akshay1y2**

### CODE REVIEW

**Comment by akshay1y2** [comment]
Tasks: https://docs.google.com/document/d/17GD0udHhXzTmRNOZZy_nsysIBnafIomr_xi4OkL-jYQ/edit?pli=1#

File: app/models/order.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
   has_many :line_items, dependent: :destroy
   belongs_to :user, optional: true
 
+  scope :by_date, ->(**args) { where(created_at: (args[:from] || (Time.now - 1.day))..(args[:to] || Time.now)) }
```
Comment (siddharth28) [inline app/models/order.rb]: not fulfilling the intended purpose.
revisit the problem statement.

File: app/models/cart.rb [hunk 2]
```diff
@@ -16,4 +16,8 @@ def add_product(product)
   def total_price
     line_items.to_a.sum { |item| item.total_price }
   end
+
+  def enabled_products
```
Comment (siddharth28) [inline app/models/cart.rb]: @akshay1y2 why have we added this?

File: app/models/order.rb [hunk 3]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
   has_many :line_items, dependent: :destroy
   belongs_to :user, optional: true
 
+  scope :by_date, ->(**args) { where(created_at: (args.empty? ? Time.now.to_date : args[:from]..args[:to])) }
```
Comment (siddharth28) [inline app/models/order.rb]: @akshay1y2 not fulfiling the purpose.

File: app/models/order.rb [hunk 4]
```diff
@@ -6,6 +6,8 @@ class Order < ApplicationRecord
   has_many :line_items, dependent: :destroy
   belongs_to :user, optional: true
 
+  scope :by_date, ->(**args) { where(created_at: (args.empty? ? Date.today.midnight..Time.now : args[:from]..args[:to])) }
```
Comment (bansalakhil) [inline app/models/order.rb:9]: @akshay1y2 Use Time.current 

Read about the difference between Time.now and Time.current

File: app/models/order.rb [hunk 5]
```diff
@@ -7,6 +7,8 @@ class Order < ApplicationRecord
   has_many :line_items, dependent: :destroy
   belongs_to :user, optional: true
 
+  scope :by_date, ->(**args) { where(created_at: (args.empty? ? Date.today.midnight..Time.current : args[:from]..args[:to])) }
```
Comment (bansalakhil) [inline app/models/order.rb:10]: Date.today => Time.current.beginning_of_day

**Pull request merged**